seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/flow/TransformFlowLifeCycle.java

```diff
@@ -89,22 +91,59 @@ public void received(Record<?> record) {
                 return;
             }
             T inputData = (T) record.getData();
-            T outputData = inputData;
-            for (SeaTunnelTransform<T> t : transform) {
-                outputData = t.map(inputData);
-                log.debug("Transform[{}] input row {} and output row {}", t, inputData, outputData);
-                if (outputData == null) {
-                    log.trace("Transform[{}] filtered data row {}", t, inputData);
-                    break;
+            List<T> outputDataList = transform(inputData);
+            if (!outputDataList.isEmpty()) {
+                // todo log metrics
+                for (T outputData : outputDataList) {
+                    collector.collect(new Record<>(outputData));
```

Comment:
@Hisoka-X I have a question. Do I need to record the metrics information of transform once before `transform` and again after the result?

Reply:
I think the metric needs to be recorded before and after each transform.

Reply:
Yes, our metrics only record source output and sink input. We should support record transform input/output too.

Reply:
get